soda/core/soda/execution/sample_tables_run.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from soda.execution.query import Query
+from soda.profiling.sample_tables_result import SampleTablesResult
+from soda.sodacl.discover_tables_cfg import DiscoverTablesCfg
+
+if TYPE_CHECKING:
+    from soda.execution.data_source_scan import DataSourceScan
+
+
+QUERY_PREFIX = "sample_tables: "
+
+
+class SampleTablesRun:
+    def __init__(self, data_source_scan: DataSourceScan, discover_tables_cfg: DiscoverTablesCfg):
+
+        self.data_source_scan = data_source_scan
+        self.soda_cloud = data_source_scan.scan._configuration.soda_cloud
+        self.data_source = data_source_scan.data_source
+        self.data_source_name = self.data_source.data_source_name
+        self.discover_tables_cfg: DiscoverTablesCfg = discover_tables_cfg
```

Comment:
cool to see this discover tables cfg abstraction 👍

Reply:
Actually, I created `SampleTablesCfg` but forget to change it :D SampleTablesCfg is exactly same with `DiscoverTablesCfg`. Maybe we can rename `DiscoverTablesCfg` to `TablesCfg` and use it in both discover table and sample table. Wdyt @bastienboutonnet @tombaeyens ?

Reply:
No objection with this idea, so go ahead and ping me to review again on this.

Reply:
good idea indeed to merge them.  consider `TableSelectionCfg` or `TableSetCfg` as a more descriptive name.